internal/expr/bindtypes.go

```diff
@@ -138,15 +138,21 @@ func bindInputTypes(ti typeNameToInfo, e *inputExpr) (tm typeinfo.Member, err er
 		}
 	}()
 
-	info, ok := ti[e.sourceType.typeName]
+	info, ok := ti[e.sourceType.getTypeName()]
 	if !ok {
-		return nil, typeMissingError(e.sourceType.typeName, getKeys(ti))
+		return nil, typeMissingError(e.sourceType.getTypeName(), getKeys(ti))
 	}
 
-	tm, err = info.TypeMember(e.sourceType.memberName)
-	if err != nil {
-		return nil, err
+	switch t := e.sourceType.(type) {
+	case memberAcessor:
+		tm, err = info.TypeMember(t.memberName)
+		if err != nil {
+			return nil, err
+		}
+	case sliceRangeAccessor:
+		return nil, fmt.Errorf("slice support not implemented")
```

Comment:
Well for this PR I wanted to make it as small as possible and to stop the flow of execution early. You are right in the future this will be a method on the `sliceRangeAccessor`. I can remove the type switching here and add the method which will only return an error.

Reply:
My idea was to limit this PR to the bare minimum to get the parsing done. In this case I think it is useful to stop execution here and, in the next PR, we can rework the type information functions (e.g. at the moment `TypeMember` is the entrypoint but it expects the name of a column, a notion which does not translate to slices).

If you agree I would like to postpone this discussion until the next PR (which Alastair is writing at the moment) and then, we can define the correct abstraction.